tests/cli-self-upd.rs

```diff
@@ -441,8 +441,8 @@ fn uninstall_removes_path_from_profile() {
 
 #[test]
 #[cfg(unix)]
-fn uninstall_removes_path_from_bash_profile() {
-    uninstall_removes_path_from_rc(".bash_profile");
+fn uninstall_removes_path_from_bashrc() {
```

Comment:
Once other review changes are taken into consideration, I think you'll probably want to add coverage rather than changing existing tests (unless I'm wrong, which I may be :)). But if so, perhaps consider migrating to focused functional tests rather than more heavyweight tests, because the overall install path has little to do with the changes you're making. Some modest test infra may be needed to support that, if you need a hand bootstrapping that, let me know.